modules/tfm/tfm/boards/common/attest_hal.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2018-2021, Arm Limited. All rights reserved.
+ * Copyright (c) 2022 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ *
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+#include "tfm_attest_hal.h"
+#include "tfm_plat_boot_seed.h"
+#include "tfm_plat_device_id.h"
+#include "tfm_plat_otp.h"
+#include <nrf_cc3xx_platform.h>
+#include "tfm_strnlen.h"
+
+/* 16 is the only supported length */
+#define CONFIG_SB_PUBLIC_KEY_HASH_LEN 16
+#include "bl_storage.h"
+
+#include <nrfx_nvmc.h>
+
+
+static enum tfm_security_lifecycle_t map_bl_storage_lcs_to_tfm_slc(lcs_t lcs){
+	switch (lcs) {
+	case ASSEMBLY:
+		return TFM_SLC_ASSEMBLY_AND_TEST;
+	case PROVISION:
+		return TFM_SLC_PSA_ROT_PROVISIONING;
+	case SECURE:
+		return TFM_SLC_SECURED;
+	case DECOMMISSIONED:
+		return TFM_SLC_DECOMMISSIONED;
+	case UNKNOWN:
+	default:
+		return TFM_SLC_UNKNOWN;
+	}
+}
+
+static lcs_t map_tfm_slc_to_bl_storage_lcs(enum tfm_security_lifecycle_t lcs){
+	switch (lcs) {
+	case TFM_SLC_ASSEMBLY_AND_TEST:
+		return ASSEMBLY;
+	case TFM_SLC_PSA_ROT_PROVISIONING:
+		return PROVISION;
+	case TFM_SLC_SECURED:
+		return SECURE;
+	case TFM_SLC_DECOMMISSIONED:
+		return DECOMMISSIONED;
+	case TFM_SLC_UNKNOWN:
+	default:
+		return UNKNOWN;
+	}
+}
+
+/* This is temporary solution until the bl_storage API is available in TF-M
+ * This points to the UICR OTP region
+ */
+/* Taken from nrf/subsys/bootloader/include/dummy_values */
+#define PM_PROVISION_ADDRESS 0xFF8100
+static const struct bl_storage_data *p_bl_storage_data =
+	(struct bl_storage_data *)PM_PROVISION_ADDRESS;
```

Comment:
I don't think this has to be a temporary solution.
```
#include "pm_config.h"
#define PM_PROVISION_ADDRESS 0xff8100
```
Unless I am missing something?

Reply:
We now take this from the bl_storage with the logic that you are mentioning I think. It should be OK now.